llvm/include/llvm/IR/BasicBlock.h

```diff
@@ -81,17 +81,17 @@ class BasicBlock final : public Value, // Basic blocks are data objects also
   /// intrinsics into DbgMarkers / DbgRecords. Deletes all dbg.values in
   /// the process and sets IsNewDbgInfoFormat = true. Only takes effect if
   /// the UseNewDbgInfoFormat LLVM command line option is given.
-  void convertToNewDbgValues();
+  void convertToNewDbgValues(bool UpdateFlagOnly = false);
```

Comment:
> I think a separate function would signal intent more clearly that only a flag is changing.
This is probably fine - one of the reasons for making it a parameter rather than a function was that in one of the cases here, we're using a non-constant as the argument here (i.e. `/*UpdateFlagOnly=*/ ModuleIsNewDbgInfoFormat || !SeenDebugRecord`); this can easily be changed to an if-else though.